src/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj

```diff
@@ -6,12 +6,33 @@
 
 (trapperkeeper/defservice metrics-service
   metrics/MetricsService
-  [[:ConfigService get-in-config]]
+  [[:ConfigService get-in-config]
+   [:WebroutingService add-ring-handler get-route]]
+
   (init [this context]
-    (core/initialize (get-in-config [:metrics] {})))
+    (let [path (get-route this)
+          handler (core/build-handler path)]
+      (add-ring-handler this handler))
+    {:registries
+      (atom {::default-registry
+            (core/initialize (get-in-config [:metrics] {}) {})})})
+
   (stop [this context]
-    (core/stop (tk-services/service-context this)))
+    (let [{:keys [registries] :as ctx} (tk-services/service-context this)]
+      (doseq [[_ metrics-reg] @registries]
+        (core/stop metrics-reg))
+      ctx))
+
+  (get-metrics-registry [this]
+    (-> @(:registries (tk-services/service-context this))
+        ::default-registry
+        :registry))
+
+  (get-metrics-registry [this registry opts]
+    (let [{:keys [registries]} (tk-services/service-context this)]
```

Comment:
It'd be sweet if you could push some or all of the rest of this function into the core namespace and use schema on the fns to make it easier to intuit what types are legal for `registry` and `opts`.